QEfficient/generation/text_generation_inference.py

```diff
@@ -269,68 +289,66 @@ def print_latency_stats_kv(
 
 
 def cloud_ai_100_exec_kv(
-    batch_size,
     tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
     qpc_path: str,
     prompt: Optional[List[str]] = None,
     device_id: List[int] = [0],
-    input_len: Optional[int] = None,
     generation_len: Optional[int] = None,
     enable_debug_logs: bool = False,
     stream: bool = True,
     write_io_dir: Optional[str] = None,
     automation=False,
 ):
+    batch_size = get_compilation_batch_size(qpc_path)
+    check_batch_size_and_num_prompts(prompt, batch_size)
+
     if batch_size == 1:
         prefill_time = []
         decode_perf = []
         total_perf = []
         total_time = []
         generated_texts = []
         for i in range(len(prompt)):
-            latency_stats = cloud_ai_100_exec_kv_helper(
+            execinfo = cloud_ai_100_exec_kv_helper(
                 tokenizer=tokenizer,
                 prompt=[prompt[i]],
                 qpc=qpc_path,
                 device_id=device_id,
-                input_len=input_len,
                 generation_len=generation_len,
                 enable_debug_logs=enable_debug_logs,
                 stream=stream,
                 write_io_dir=write_io_dir,
             )
-            generated_texts.append(latency_stats[0])
-            prefill_time.append(latency_stats[1])
-            decode_perf.append(latency_stats[2])
-            total_perf.append(latency_stats[3])
-            total_time.append(latency_stats[4])
+            generated_texts.append(execinfo.generated_texts)
+            prefill_time.append(execinfo.prefill_perf)
+            decode_perf.append(execinfo.decode_perf)
+            total_perf.append(execinfo.total_perf)
+            total_time.append(execinfo.total_time)
 
         prefill_time = np.average(prefill_time)
         decode_perf = np.average(decode_perf)
         total_perf = np.average(total_perf)
         total_time = np.average(total_time)
 
     else:
-        latency_stats = cloud_ai_100_exec_kv_helper(
+        execinfo = cloud_ai_100_exec_kv_helper(
             tokenizer=tokenizer,
             prompt=prompt,
             qpc=qpc_path,
             device_id=device_id,
-            input_len=input_len,
             generation_len=generation_len,
             enable_debug_logs=enable_debug_logs,
             stream=stream,
             write_io_dir=write_io_dir,
         )
-        generated_texts, prefill_time, decode_perf, total_perf, total_time = latency_stats
-
     print_latency_stats_kv(
         prompt,
-        generated_texts,
-        batch_size,
-        prefill_time,
-        decode_perf,
-        total_perf,
-        total_time,
+        generated_texts=execinfo.generated_texts,
```

Comment:
pass `prompt, execinfo, automation` handle printing the attributes inside `print` function.

Reply:
updated